modin/utils.py

```diff
@@ -576,20 +567,19 @@ def try_cast_to_pandas(obj: Any, squeeze: bool = False) -> Any:
     object
         Converted object.
     """
-    if isinstance(obj, SupportsPrivateToPandas):
-        result = obj._to_pandas()
-        if squeeze:
-            result = result.squeeze(axis=1)
-        return result
     if isinstance(obj, SupportsPublicToPandas):
         result = obj.to_pandas()
         if squeeze:
             result = result.squeeze(axis=1)
-        # Query compiler case, it doesn't have logic about convertion to Series
+
+        from modin.core.storage_formats import BaseQueryCompiler
+
         if (
-            isinstance(getattr(result, "name", None), str)
+            isinstance(obj, BaseQueryCompiler)
```

Comment:
Why do we need this additional condition?

Reply:
Previously, this logic was only applied to objects having public `.to_pandas()` method (query compilers, low-level modin frames), so I thought that putting this condition is necessary. But on the second though, it seems that this logic works just fine for high-level dataframes as well, so removed the condition in the last commit